### PR TITLE
Rename campaign.influencer to campaign.user and use foreign key.

### PR DESCRIPTION
**Pull request by tjgupta**
For clarity I'd like to call user "user" rather than "influencer". As the first part of this change I'm renaming the "foreign key" (or as much as Mongo has notion of a relational structure) on Campaign and Event to "user" and using the Mongoose foreign relationship schema style so we get the benefits of that later (should we need a "join").

Note that this is a backward incompatible data model change, which is why I'd like to do it before we roll this out. I plan to rename the Influencer model too in a soon to follow PR.

### CODE REVIEW

File: app/models/campaign.js [hunk 1]
```diff
@@ -32,29 +33,25 @@ CampaignSchema.statics.findForRequestingUser = function(campaignId, requestUserI
                 return null;
             }
             // if campaign is not published, only return to owner of campaign
-            if (!campaign.publish && campaign.influencer !== requestUserId) {
+            if (!campaign.publish && campaign.user.toString() !== requestUserId) {
```
Comment (WCRest) [inline app/models/campaign.js]: Shouldn't this be requestUser instead of requestUserId now?

**Comment by WCRest** [comment]
There is a lot of test data on the staging database, and I think it'd be pretty straightforward to update the existing data to match the new schema. It might be worth it for the convenience of everyone testing on staging.

**Comment by tjgupta** [comment]
We could update the test data. Although my next PR was going to rename the entire Influencer collection, but perhaps that's also easy to do in mongo. I just wanted to make sure to get these changes out before we go live so we're not stuck with the concept of Influencer vs. User down the road.

Because if we some day want to allow users to be able to sign up (for following, commenting on a campaign, etc) but not necessarily have permission to create a campaign, this is where our current "Influencer" naming gets super weird.

**Comment by tjgupta** [comment]
I added the second part to this same PR (renaming the Influencer model to User). It's actually a pretty small change, pretty much just find/replace on the word influencer.

Also, renaming the influencers collection to users does work, with the caveat that there could be some funky lurking data related bugs in there. And the other caveat related to the first change is that campaigns created before this PR won't show up properly since they have the "influencer" foreign key rather than "user", unless we decide to migrate the schema.

https://docs.mongodb.com/manual/reference/method/db.collection.renameCollection/

**Pull request merged**